### PR TITLE
Dual-stack: /healthz/ready should have both v4/v6 wildcard listeners

### PR DESCRIPTION
**Pull request by kfaseela**
**Please provide a description of this PR:**

- Enhance DualStack integration tests as per [RFC](https://docs.google.com/document/d/1sou7tc52HZsH2vRjcuP-N7VvtvWH-7_f/edit) - added gateway readiness probe tests
- Fix issue with probe test as mentioned in https://github.com/istio/istio/issues/47013

### CODE REVIEW

File: tests/integration/pilot/gateway_test.go [hunk 1]
```diff
@@ -563,3 +563,36 @@ func StatusGatewayTest(t framework.TestContext) {
 	// It should be added back
 	retry.UntilSuccessOrFail(t, check)
 }
+
+// Verify that the envoy readiness probes are reachable at
+// https://GatewayPodIP:15021/healthz/ready . This is being explicitly done
+// to make sure, in dual-stack scenarios both v4 and v6 probes are reachable.
+func TestGatewayReadinessProbes(t *testing.T) {
```
Comment (howardjohn) [inline tests/integration/pilot/gateway_test.go:570]: you can call the gw Service, 15021 is exposed in the Service.

I think?

**Review by howardjohn — COMMENTED** [review]
LGTM, leaving for the dual stack experts to review though.

Nice fix

File: tests/integration/pilot/gateway_test.go (hunk 1 above)
Comment (kfaseela) [reply tests/integration/pilot/gateway_test.go:570]: https://github.com/istio/istio/pull/44024 is not yet merged, and that is another bug that gateway svc is not deployed as a dual-stack service.

**Review by jacob-delgado — APPROVED** [review]
great find!

File: tools/packaging/common/envoy_bootstrap.json [hunk 2]
```diff
@@ -550,6 +550,19 @@
              "port_value": {{ .envoy_status_port }}
            }
         },
+        {{- if .dual_stack }}
+        "additionalAddresses": [
```
Comment (adams-shaun) [inline tools/packaging/common/envoy_bootstrap.json:554]: should the metrics endpoint above also listen on additional address?

Comment (kfaseela) [reply tools/packaging/common/envoy_bootstrap.json:554]: The current PR is only for /healthz/ready, with integration tests added for the same. I intend to add integration tests for metrics in a separate PR, and at that time, the bugs related to that can be fixed. Don't want to put everything into a single PR

Comment (adams-shaun) [reply tools/packaging/common/envoy_bootstrap.json:554]: sounds great, thanks

**Review by adams-shaun — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Review by zhlsunshine — APPROVED** [review]
LGTM

File: pkg/bootstrap/config.go [hunk 3]
```diff
@@ -153,15 +153,18 @@ func (cfg Config) toTemplateParams() (map[string]any, error) {
...
+					option.AdditionalWildCard(option.WildcardIPv4),
 					option.DNSLookupFamily(option.DNSLookupFamilyIPS))
 			} else {
 				opts = append(opts,
 					option.Localhost(option.LocalhostIPv4),
 					option.Wildcard(option.WildcardIPv4),
+					option.AdditionalWildCard(option.WildcardIPv6),
 					option.DNSLookupFamily(option.DNSLookupFamilyIPS))
 			}
+			opts = append(opts, option.DualStack(true))
```
Comment (hzxuzhonghu) [inline pkg/bootstrap/config.go:165]: What's the result if dual stack is enabled on a non dual stack cluster? Will the envoy fail to start the listener?

Comment (hzxuzhonghu) [reply pkg/bootstrap/config.go:165]: MVM, this is already on the reality that proxy support dual stack

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**